CefSharp.Core/Internals/CookieManager.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright � 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#include "Stdafx.h"
+#include "CookieManager.h"
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        void CookieManager::ThrowIfDisposed()
+        {
+            if (_cookieManager.get() == nullptr)
+            {
+                throw gcnew ObjectDisposedException("CookieManager");
+            }
+        }
+
+        Task<bool>^ CookieManager::DeleteCookiesAsync(String^ url, String^ name)
+        {
+            ThrowIfDisposed();
+
+            auto cookieInvoker = gcnew CookieAsyncWrapper(_cookieManager.get(), url, name);
+
+            if (CefCurrentlyOn(TID_IO))
+            {
+                auto source = gcnew TaskCompletionSource<bool>();
+                source->TrySetResult(cookieInvoker->DeleteCookies());
+                return source->Task;
+            }
+
+            auto task = Cef::IOThreadTaskFactory->StartNew(gcnew Func<bool>(cookieInvoker, &CookieAsyncWrapper::DeleteCookies));
+
+            delete cookieInvoker;
```

Comment:
Is the `CookieAsyncWrapper` disposed of before the task completes?

Might be safer for `CookieAsyncWrapper` to release the `CookieManager` reference once it's made the call, then return the result. Thoughts?